src/classes/BDI_DataImportService.cls

```diff
@@ -631,6 +639,29 @@ global with sharing class BDI_DataImportService {
         return fieldAccessLevelsToValidate;
     }
 
+    /**
+     * @description Before processing data import records, this method filters out any data imports that do not meet
+     * custom criteria that applies to all data imports before allowing them to be processed.
+     * @param dataImports {List<DataImport__c>} List of data import records to be filtered
+     *
+     * @return List<DataImport__c>
+    */
+    private List<DataImport__c> shouldProcess(List<DataImport__c> dataImports) {
```

Comment:
This approach - filtering the list - definitely works, but I'm wondering if we should have this method check one data import at a time, and then return a boolean based on whether it should be processed or not. I know I suggested this name, but we should either update this to include more of criteria about when we should process a record, or name it more specific to payment status. Then this method could be called at the very beginning of the processing loop - line #917 in my file, but right where the comment "// skip di's that already have an error" is currently. The main advantage to handling it that way is that it avoids additional iterations over the list of data imports, and we are very focused on performance at scale here. 

IF we end up keeping this implementation - and I think I'm leaning more towards the one-record-check version - we should rename this to something like "filterPendingPayments" or something more descriptive

but I'm wondering if we should either rename this method to "

Reply:
UPDATE: After further conversation, I'm back on board with the iteration approach - iterating through and filtering the DI list based on payment status.